### PR TITLE
Dropdown task editor add tests, convert to ES6

### PR DESCRIPTION
**Pull request by mcbouslog**
Staging branch URL: https://dropdown-es6-editor.pfe-preview.zooniverse.org/

Adds test for dropdown editor, converts to ES6.

Small step towards #3636.

No rush on this.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [x] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by mcbouslog** [comment]
refactoring to not mutate props

**Comment by mcbouslog** [comment]
not sure handling `this.props.task.selects` properly. considered using `.slice()`, but would just shallow copy when I think deep copy needed, could use `JSON.parse(JSON.stringify(this.props.task.selects))`, but maybe not necessary.

**Comment by eatyourgreens** [comment]
Sorry, this dropped off my radar. It probably needs rebasing now, as it's quite out-of-date with master.

These dropdown task structures can get quite large, which I think would cause performance problems if you try to deep copy them using `JSON.parse(JSON.stringify())`.

**Comment by mcbouslog** [comment]
No issues with a rebase and I updated the staged branch.

I started a conversion for the related dropdown-dialog and dropdown-list components that I'll revisit now, to complete the conversion. 

Then I'm thinking I'll refactor all dropdown related components in light of React 16 lifecycle recommendations (i.e. update on prop as key).

File: app/classifier/tasks/dropdown/editor.jsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+              onReorder={this.updateSelects}
+              editDropdown={this.editDropdown}
+              deleteDropdown={this.deleteDropdown}
+            />
+            <p>
+              <button type="button" className="minor-button" onClick={() => this.editDropdown('create')}>
+                <i className="fa fa-plus" />
+                Add a Dropdown
+              </button>
+              <label htmlFor="condition">
```
Comment (eatyourgreens) [inline app/classifier/tasks/dropdown/editor.jsx]: `htmlFor` is redundant here, since the label wraps the form control.

**Comment by eatyourgreens** [comment]
Couple of minor points about form labels but I'm happy to merge this.

**Comment by mcbouslog** [comment]
Good suggestions, thanks!

**Pull request merged**